src/sentry/grouping/api.py

```diff
@@ -205,61 +264,39 @@ def load_default_grouping_config() -> StrategyConfiguration:
     return load_grouping_config(config_dict=None)
 
 
-def get_fingerprinting_config_for_project(
-    project: Project, config_id: str | None = None
-) -> FingerprintingRules:
-    """
-    Returns the fingerprinting rules for a project.
-    Merges the project's custom fingerprinting rules (if any) with the default built-in rules.
-    """
-
-    from sentry.grouping.fingerprinting import FingerprintingRules, InvalidFingerprintingConfig
-
-    if features.has("organizations:grouping-built-in-fingerprint-rules", project.organization):
-        bases = get_projects_default_fingerprinting_bases(project, config_id=config_id)
-    else:
-        bases = []
-    rules = project.get_option("sentry:fingerprinting_rules")
-    if not rules:
-        return FingerprintingRules([], bases=bases)
-
-    from sentry.utils.cache import cache
-    from sentry.utils.hashlib import md5_text
-
-    cache_key = "fingerprinting-rules:" + md5_text(rules).hexdigest()
-    rv = cache.get(cache_key)
-    if rv is not None:
-        return FingerprintingRules.from_json(rv, bases=bases)
-
+def apply_server_fingerprinting(
+    event_data: CanonicalKeyDict,
+    grouping_config: GroupingConfig,
+    # TODO: determine if we can pass around FingerprintingRules instance instead of serializing and deserializing it back and forth
+    allow_custom_title: bool = True,
+) -> None:
     try:
-        rv = FingerprintingRules.from_config_string(rules, bases=bases)
-    except InvalidFingerprintingConfig:
-        rv = FingerprintingRules([], bases=bases)
-    cache.set(cache_key, rv.to_json())
-    return rv
-
-
-def apply_server_fingerprinting(event, config, allow_custom_title=True):
-    client_fingerprint = event.get("fingerprint")
-    rv = config.get_fingerprint_values_for_event(event)
+        config = FingerprintingRules.from_json(grouping_config["fingerprinting"])
+    except KeyError:
+        # handle the case we're getting a legacy config serialized without fingerprinting rules
+        # TODO: determine if this ever actually gets hit
```

Comment:
I sometimes use `capture_message` or `logger.error` to capture errors so I can investigate it in Sentry.